File: frontend/src/types/quiz.ts
export type LanguageCode = "en" | "vi";

export interface QuizQuestion {
  id: number;
  lessonId: number;
  difficulty: number;
  language: LanguageCode;
  prompt: string;
  options: string[];
  correctOption: string;
  reference?: string | null;
  explanation?: string | null;
}

export interface QuizProgressState {
  answeredCount: number;
  correctCount: number;
  saving: boolean;
  error: string | null;
}

File: backend/api/serializers/quiz.py
from __future__ import annotations

from backend.services.dto import LocalizedQuizQuestionDTO


def serialize_quiz(quiz: LocalizedQuizQuestionDTO) -> dict[str, object]:
    return {
        "id": quiz.id,
        "lesson_id": quiz.lesson_id,
        "difficulty": quiz.difficulty,
        "language": quiz.language,
        "prompt": quiz.prompt,
        "options": quiz.options,
        "correct_option": quiz.correct_option,
        "reference": quiz.reference,
        "explanation": quiz.explanation,
    }
